Improve command-line usage

tyrannosaurus-becks opened the pull request:
This PR adds a user-friendly **description** field for each config parameter. This results in an improvement to the command-line help experience, which formerly only listed the names of the fields with no further description.

This PR is intended to be included in 0.2.0.

```
➜  authenticator git:(config-help-improvements) ✗ authenticator -h
Usage of authenticator:

listener
  -host string
      The address at which Approzium's authenticator should run.
  -http-port int
      The port that should serve HTTP traffic.
  -grpc-port int
      The port that should serve GRPC traffic, used for client authentication requests.

tls
  -disable-tls bool
      Set to true to disable TLS encryption. Not recommended in production environments.
  -tls-cert-path string
      The path to the certificate proving the Approzium authenticator's identity.
  -tls-key-path string
      The path to the private key to be used for decrypting inbound communication.

logging
  -log-level string
      Valid values are "panic", "fatal", "error", "warn", "info", "debug", and "trace".
  -log-format string
      Valid values are "text" and "json".
  -log-raw bool
      Set to true to disable removing sensitive values from logging.

secrets
  -secrets-manager string
      Valid values include "vault" for HashiCorp Vault, "asm" for AWS Secrets Manager", or "local" for a file-based development environment.
  -vault-token string
      If "vault" is selected as the secrets manager, a Vault token that won't expire for the Approzium authenticator's lifetime. Use "vault token path" instead in production.
  -vault-token-path string
      If "vault" is selected as the secrets manager, the path to a file containing a Vault token that is constantly refreshed by the Vault agent.
  -vault-addr string
      If "vault" is selected as the secrets manager, Vault's address.
  -assume-aws-role string
      If "asm" is selected as the secrets manager, an optional role for the Approzium authenticator to assume when communicating with AWS.
  -aws-region string
      If "asm" is selected as the secrets manager, the region where the AWS Secrets Manager resides.

miscellaneous
  -config string
      The path to a yaml file bearing configuration settings.
  -dev bool
      When true, starts the Approzium authenticator in dev mode, with TLS disabled, with the "local" secrets manager, and debug-level logs.
  -version bool
      Outputs the Approzium authenticator's version.
```

UpGado commented:
Elegant code 💯. Is there a way to have the parameter order be the same as in the website?

tyrannosaurus-becks commented:
@UpGado - hmm, good idea. [It looks like it might be possible](https://golang.org/pkg/flag/#FlagSet) if we used a `FlagSet` and defined our own `Usage` function. I'll try it out! 😄

tyrannosaurus-becks commented:
@UpGado I went ahead and updated how the output looks - both by group, and in the order it's included in the registry. I've updated the example in the PR description to match what it looks like now - please see the example at the top. Looks much nicer!

tyrannosaurus-becks commented:
I went ahead and merged this so it can be in the 0.2.0 release.
